### PR TITLE
Clean up file locations

### PR DESCRIPTION
**Pull request by LukeShu**
## Description

There are too many directories containing auxiliary build scripts. The root directory is getting cluttered.

I didn't touch the `releng/` directory, because it is its own can of worms, but otherwise I centralized a bunch of files in to `build-aux/` (from `builder/` and `tools/build/`)

I moved a bunch of "CAPS" `.md` files in to directories `Community/` and `DevDocumentation/`, and added pointers to them to the main `README.md`. In Unix tradition, I named these starting with capital letters, because things with capital letters are things you aught to pay attention to (and aught to be show first, but sadly the GH webui doesn't do this, and (unless you set `LC_COLLATE=C`) neither does `ls` on many modern boxes).  `Documentation/` is a common name (used by some obscure projects such as Git and Linux), but I didn't want to use it because I didn't want to conflate dev docs with user docs; so I added the `Dev`-prefix.

I moved the GNU Make primer from a comment in `builder.mk` in to a separate `DevDocumentation/gnu-make.md`.

## Related Issues

(none)

## Testing

I grepped for links that would be broken and fixed them; but it's possible I missed something.

## Checklist

- [x] **Does my change need to be backported to a previous release?** - no
  - What backport versions were discussed with the Maintainers in the Issue?

- [x] **I made sure to update `CHANGELOG.md`.** - no applicable changes

   Remember, the CHANGELOG needs to mention:
  - Any new features
  - Any changes to our included version of Envoy
  - Any non-backward-compatible changes
  - Any deprecations

- [x] **This is unlikely to impact how Ambassador performs at scale.** - not a runtime change

   Remember, things that might have an impact at scale include:
  - Any significant changes in memory use that might require adjusting the memory limits
  - Any significant changes in CPU use that might require adjusting the CPU limits
  - Anything that might change how many replicas users should use
  - Changes that impact data-plane latency/scalability

- [ ] **My change is adequately tested.**

   Remember when considering testing:
  - Your change needs to be specifically covered by tests.
    - Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
  - You also need to make sure that the _entire area being changed_ has adequate test coverage.
    - If existing tests don't actually cover the entire area being changed, add tests.
    - This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
  - We should lean on the bulk of code being covered by unit tests, but...
  - ... an end-to-end test should cover the integration points

- [x] **I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.** - yep, I updated references in it

- [ ] **The changes in this PR have been reviewed for security concerns and adherence to security best practices.**

### CODE REVIEW

File: DevDocumentation/gnu-make.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      $@ = the target
+      $^ = the list of dependencies (space separated)
+      $< = the first (left-most) dependency
+      $* = the value of the % glob in a pattern rule
+
+      Each of these have $(@D) and $(@F) variants that are the
+      directory-part and file-part of each value, respectively.
+
+      I think those are easy enough to remember mnemonically:
+        - $@ is where you shoul direct the output at.
```
Comment (kflynn) [inline DevDocumentation/gnu-make.md:38]: `s/shoul/should/` when you get a chance.

File: DevDocumentation/gnu-make.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+      # is the same as
+
+      target1: deps
+      	command to run
+      target2: deps
+      	command to run
+
+  Because of this, if you have a command that generates multiple,
+  outputs, it _must_ be a pattern rule:
```
Comment (kflynn) [inline DevDocumentation/gnu-make.md:64]: `s/multiple, outputs/multiple outputs/` when you get a chance.

File: DevDocumentation/gnu-make.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      target: normal-deps | order-only-deps
+
+  Dependencies after the "|" are created if they don't exist, but if
+  they already exist, then don't bother updating them.
+
+Tips:
+-----
+
+ - Use absolute filenames.  It's dumb, but it really does result in
+   fewer headaches.  Use $(OSS_HOME) and $(AES_HOME) to spell the
```
Comment (kflynn) [inline DevDocumentation/gnu-make.md:82]: Do these still exist?

**Review by kflynn — APPROVED** [review]
I think that this is a good idea. I left a few small nits for whenever you get to them, but I'm approving as-is.

File: DevDocumentation/gnu-make.md (hunk 1 above)
Comment (LukeShu) [reply DevDocumentation/gnu-make.md]: Fixed.

File: DevDocumentation/gnu-make.md (hunk 2 above)
Comment (LukeShu) [reply DevDocumentation/gnu-make.md]: Fixed.

File: DevDocumentation/gnu-make.md (hunk 3 above)
Comment (LukeShu) [reply DevDocumentation/gnu-make.md]: This tip is less relevant to Emissary these days, but I don't quite want to remove it. I've added some more discussion to the tip.

Comment (kflynn) [reply DevDocumentation/gnu-make.md]: 👍

**Review by LanceEa — APPROVED** [review]
lgtm..thanks!

**Review by haq204 — APPROVED** [review]

**Pull request merged**